src/liquid/components/ld-tabs/test/ld-tabs.spec.ts

```diff
@@ -0,0 +1,665 @@
+      const ldTabpanellist = ldTabs.querySelector('ld-tabpanellist')
+      const ldTabpanels = ldTabpanellist.querySelectorAll('ld-tabpanel')
+      expect(ldTabpanels.length).toEqual(4)
+
+      const tabBtn0 = ldTabItems[0].querySelector('button')
+      const tabBtn3 = ldTabItems[3].querySelector('button')
+
+      expect(tabBtn0.getAttribute('aria-selected')).toEqual('true')
+      expect(tabBtn0.getAttribute('tabindex')).toEqual(null)
+
+      const ldTabpanel0 = ldTabpanels[0]
+      expect(ldTabpanel0.classList.contains('ld-tabpanel--hidden')).toEqual(
+        false
+      )
+
+      expect(tabBtn3.getAttribute('aria-selected')).toEqual(null)
+      expect(tabBtn3.getAttribute('tabindex')).toEqual('-1')
+
+      const ldTabpanel3 = ldTabpanels[3]
+      expect(ldTabpanel3.classList.contains('ld-tabpanel--hidden')).toEqual(
+        true
+      )
+
+      ldTabItems[3].scrollIntoView = jest.fn()
+      const spyScrollIntoView = jest.spyOn(ldTabItems[3], 'scrollIntoView')
+
+      tabBtn3.dispatchEvent(new Event('click'))
+      await page.waitForChanges()
+
+      expect(tabBtn0.getAttribute('aria-selected')).toEqual('true')
+      expect(tabBtn0.getAttribute('tabindex')).toEqual(null)
+
+      expect(ldTabpanel0.classList.contains('ld-tabpanel--hidden')).toEqual(
+        false
+      )
+
+      expect(tabBtn3.getAttribute('aria-selected')).toEqual(null)
+      expect(tabBtn3.getAttribute('tabindex')).toEqual('-1')
+
+      expect(ldTabpanel3.classList.contains('ld-tabpanel--hidden')).toEqual(
+        true
+      )
+
+      expect(spyScrollIntoView).not.toHaveBeenCalled()
+    })
+
+    it('does not change tab via click on already selected tab', async () => {
+      const page = await newSpecPage({
+        components,
+        html: `
+        <ld-tabs>
+          <ld-tablist>
+            <ld-tab selected>Fruits</ld-tab>
+            <ld-tab>Vegetables</ld-tab>
+            <ld-tab>Nuts</ld-tab>
+            <ld-tab disabled>Grain</ld-tab>
+          </ld-tablist>
+        </ld-tabs>
+      `,
+      })
+      const ldTabs = page.root
+      const ldTablist = ldTabs.querySelector('ld-tablist')
+
+      const ldTabItems = ldTablist.querySelectorAll('ld-tab')
+      expect(ldTabItems.length).toEqual(4)
+
+      const tabBtn0 = ldTabItems[0].querySelector('button')
+
+      expect(tabBtn0.getAttribute('aria-selected')).toEqual('true')
+      expect(tabBtn0.getAttribute('tabindex')).toEqual(null)
+
+      ldTabItems[0].scrollIntoView = jest.fn()
+      const spyScrollIntoView = jest.spyOn(ldTabItems[0], 'scrollIntoView')
+
+      tabBtn0.dispatchEvent(new Event('click'))
+      await page.waitForChanges()
+
+      expect(tabBtn0.getAttribute('aria-selected')).toEqual('true')
+      expect(tabBtn0.getAttribute('tabindex')).toEqual(null)
+
+      expect(spyScrollIntoView).not.toHaveBeenCalled()
+    })
+  })
+
+  describe('keyboard interactions', () => {
+    it('changes focus via arrow keys and selection via enter or space', async () => {
```

Comment:
Every (keyboard) action should have its own test.

Reply:
Yeah, I know. But in this case the right button does pretty much the same as the left one and then I just need to hit the enter button etc. so I decided to https://youtu.be/v9wIY76Jfu8?t=39